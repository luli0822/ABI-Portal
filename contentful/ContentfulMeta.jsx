import React, { useContext } from 'react';
import PropTypes from 'prop-types';
import Head from 'next/head';
import siteDataContext from './siteDataContext';

function ContentfulMeta({ pageId, siteDomain, metaTitle }) {
	const iconSizes = [192, 96, 32, 16];
	const appleTouchIconSizes = [180, 152, 144, 114, 120, 76, 60, 57];
	const { fieldOrFallback, pathMap, siteEntry, linkedEntries, linkedAssets, lang } = useContext(
		siteDataContext
	);

	const [pagePath] = Object.entries(pathMap).find(
		([path, options]) =>
			options.query.pageId === linkedEntries[pageId].sys.id && options.query.lang === lang
	);

	const pageUrl = siteDomain && pagePath ? `${siteDomain}${pagePath}/` : null;
	const pageEntryFields = linkedEntries[pageId].fields;
	const siteEntryFields = siteEntry.fields;
	const isIndexPage = fieldOrFallback(siteEntryFields.indexPage).sys.id === pageId;
	const siteTitle = fieldOrFallback(siteEntryFields.title);
	const favicon = fieldOrFallback(siteEntryFields.favicon);
	const faviconUrl = favicon ? fieldOrFallback(linkedAssets[favicon.sys.id].fields.file).url : '';
	const pageTitle = fieldOrFallback(pageEntryFields.title);
	const description = fieldOrFallback(pageEntryFields.description);
	const keywords = fieldOrFallback(pageEntryFields.keywords);
	const pageKeywords = keywords ? fieldOrFallback(pageEntryFields.keywords).toString() : '';
	const ogDescription = fieldOrFallback(pageEntryFields.openGraphDescription);
	const ogTitle = fieldOrFallback(pageEntryFields.openGraphTitle);
	const ogImage = fieldOrFallback(pageEntryFields.openGraphImage);
	const ogImgUrl =
		(ogImage &&
			fieldOrFallback(linkedAssets[ogImage.sys.id].fields.file) &&
			fieldOrFallback(linkedAssets[ogImage.sys.id].fields.file).url) ||
		'';

	const title =
		metaTitle ||
		(isIndexPage && siteTitle && siteTitle) ||
		(pageTitle && siteTitle && `${pageTitle} | ${siteTitle}`) ||
		pageTitle ||
		'';

	return (
		<Head>
			{/* Page title */}
			<title>{title}</title>

			{/* Page desciption */}
			{description && <meta name="description" content={description} />}

			{/* OG page url and canonical url  */}
			{pageUrl && <meta property="og:url" content={pageUrl} />}
			{pageUrl && <link rel="canonical" href={pageUrl} />}

			{/* Get the keywords as an array */}
			{keywords && <meta name="keywords" content={pageKeywords} />}

			{/* Page OG Description */}
			{ogDescription && <meta property="og:description" content={ogDescription} />}

			{/* Page OG title */}
			{ogTitle && <meta property="og:title" content={ogTitle} />}

			{/* Page OG image */}
			{ogImage && <meta property="og:image" content={ogImgUrl} />}

			{/* Favicon links */}
			{faviconUrl && <link rel="icon" type="image/png" href={`${faviconUrl}?fm=png`} />}
			{faviconUrl &&
				iconSizes.map((size, index) => (
					<link
						rel="icon"
						type="image/png"
						sizes={`${size}x${size}`}
						href={`${faviconUrl}?w=${size}&fm=png`}
						// eslint-disable-next-line react/no-array-index-key
						key={index}
					/>
				))}
			{faviconUrl &&
				appleTouchIconSizes.map((size, index) => (
					<link
						rel="apple-touch-icon"
						sizes={`${size}x${size}`}
						href={`${faviconUrl}?w=${size}&fm=png`}
						// eslint-disable-next-line react/no-array-index-key
						key={index}
					/>
				))}
			{faviconUrl && <meta name="msapplication-TileImage" content={`${faviconUrl}?fm=png`} />}
			{siteTitle && <meta name="apple-mobile-web-app-title" content={siteTitle} />}
		</Head>
	);
}

ContentfulMeta.propTypes = {
	pageId: PropTypes.string.isRequired,
	siteDomain: PropTypes.string,
	metaTitle: PropTypes.string,
};

ContentfulMeta.defaultProps = {
	siteDomain: undefined,
	metaTitle: undefined,
};

export default ContentfulMeta;
