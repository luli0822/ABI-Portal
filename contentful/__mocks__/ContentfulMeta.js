export const mockSiteMeta = {
	pageId: 'site-home-page',
	siteDomain: 'https://www.test-fr.ca/',
};

export const mockPageMeta = {
	pageId: 'about-us-page',
	siteDomain: 'https://www.test.ca',
};

export const mockNoFavicon = {
	sys: {
		id: 'siteId',
		contentType: {
			sys: {
				id: 'site',
				type: 'link',
			},
		},
	},
	fields: {
		title: {
			'en-CA': 'en title',
			'fr-CA': 'fr title',
		},
		indexPage: {
			'en-CA': {
				sys: {
					id: 'site-home-page',
				},
				fields: {
					slug: {
						'en-CA': 'index-slug',
					},
					title: {
						'en-CA': 'index title',
					},
				},
			},
		},
		pages: {
			'en-CA': [
				{
					sys: {
						id: 'about-us-page',
					},
				},
			],
		},
	},
};
