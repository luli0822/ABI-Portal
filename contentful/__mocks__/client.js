import fetchSiteData from './fetchSiteData.json';
// Refresh mock siteData will disclude about-us
module.exports = {
	getClient: () => {},
	fetchSiteData: () => fetchSiteData,
};
