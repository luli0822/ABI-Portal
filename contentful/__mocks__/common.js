// A method to build Link data structure
export const linkDataMock = (type, linkId = 'link-id-1') => ({
	sys: {
		id: linkId,
		contentType: {
			sys: {
				type: 'Link',
				linkType: 'ContentType',
				id: `${type}`,
			},
		},
	},
	fields: {
		title: {
			'en-CA': 'Link Text',
		},
		accessibleLabel: {
			'en-CA': 'Link accessible Text',
		},
		url: type === 'externalLink' ? { 'en-CA': 'https://loblawdigital.co' } : undefined,
		page:
			type === 'pageLink'
				? {
						'en-CA': {
							sys: {
								type: 'Link',
								linkType: 'Entry',
								id: 'site-home-page',
							},
						},
				  }
				: undefined,
	},
});

// Use this for pageLink structure mock
export const pageLinkMock = linkDataMock('pageLink');

// Use this externalLink structure mock
export const externalLinkMock = linkDataMock('externalLink');

// A method to build Media/image data structure
export const imageDataMock = type => ({
	sys: {
		type: 'Asset',
		id: type === 'image/png' ? 'asset-id-1' : 'asset-id-2',
	},
	fields: {
		title: {
			'en-CA': 'Image Title Text',
		},
		description: {
			'en-CA': 'Image description Text',
		},
		file: {
			'en-CA': {
				url: 'image-file-url.png',
				details: {
					size: 150795,
					image: {
						width: 300,
						height: 300,
					},
				},
				fileName: 'png-example.png',
				contentType: type,
			},
		},
	},
});
