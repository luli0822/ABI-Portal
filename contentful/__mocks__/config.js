module.exports = {
	space: 'spaceId',
	isPreview: true,
	accessToken: 'sometoken',
	host: 'preview.contentful.com',
	environment: 'master',
	siteSlug: 'site-slug',
	defaultLang: 'en-CA',
	fallbackLang: 'en-CA',
};
