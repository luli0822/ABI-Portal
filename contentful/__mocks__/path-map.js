module.exports = {
	pathMap: {
		'/': {
			page: '/',
			query: {
				lang: 'en-CA',
				pageId: 'site-home-page',
			},
		},
		'/home-pg-fr': {
			page: '/',
			query: {
				lang: 'fr-CA',
				pageId: 'site-home-page',
			},
		},
		'/about-pg-en': {
			page: '/',
			query: {
				lang: 'en-CA',
				pageId: 'about-us-page',
			},
		},
		'/about-pg-fr': {
			page: '/',
			query: {
				lang: 'fr-CA',
				pageId: 'about-us-page',
			},
		},
		// This path map for testing refreshed index page
		'/refreshed-page': {
			page: '/',
			query: {
				lang: 'en-CA',
				pageId: 'REFRESHED-site-home-page',
			},
		},
	},
};
