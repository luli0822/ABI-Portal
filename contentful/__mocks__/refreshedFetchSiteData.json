{
  "siteEntry": {
    "sys": {
      "space": {
        "sys": {
          "id": "spaceId"
        }
      },
      "id": "siteId",
      "contentType": {
        "sys": {
          "id": "site",
          "type": "link"
        }
      }
    },
    "fields": {
      "title": {
        "en-CA": "en title",
        "fr-CA": "fr title"
      },
      "slug": {
        "en-CA": "site-slug"
      },
      "indexPage": {
        "en-CA": {
          "sys": {
            "id": "REFRESHED-site-home-page"
          }
        }
      },
      "pages": {
        "en-CA": []
      },
      "keyValues": {
        "en-CA": [
          {
            "sys": {
              "type": "Link",
              "linkType": "Entry",
              "id": "lang-toggle-key-value"
            }
          },
          {
            "sys": {
              "type": "Link",
              "linkType": "Entry",
              "id": "rich-text-key-value"
            }
          }
        ]
      },
      "navigation": {
        "en-CA": [
          {
            "sys": {
              "type": "Link",
              "linkType": "Entry",
              "id": "some-external-link"
            }
          }
        ]
      }
    }
  },
  "linkedEntries": {
    "REFRESHED-site-home-page": {
      "sys": {
        "id": "REFRESHED-site-home-page",
        "space": {},
        "contentType": {
          "id": "page"
        }
      },
      "fields": {
        "title": {
          "en-CA": "title en",
          "fr-CA": "title fr"
        },
        "description": {
          "en-CA": "description en",
          "fr-CA": "description en"
        },
        "openGraphTitle": {
          "en-CA": "openGraphTitle en",
          "fr-CA": "openGraphTitle en"
        },
        "openGraphDescription": {
          "en-CA": "openGraphDescription en",
          "fr-CA": "openGraphDescription en"
        },
        "slug": {
          "en-CA": "home-pg-en",
          "fr-CA": "home-pg-fr"
        }
      }
    },
    "some-external-link": {
      "sys": {
        "id": "some-external-link",
        "space": {},
        "contentType": {
          "sys": {
            "id": "externalLink"
          }
        }
      },
      "fields": {
        "title": {
          "en-CA": "External link text",
          "fr-CA": "External link text fr"
        },
        "url": {
          "en-CA": "https://test.com",
          "fr-CA": "https://test.com/fr"
        }
      }
    },
    "lang-toggle-key-value": {
      "sys": {
        "id": "lang-toggle-key-value",
        "space": {},
        "contentType": {
          "id": "keyValue"
        }
      },
      "fields": {
        "slug": {
          "en-CA": "lang-toggle-slug"
        },
        "value": {
          "en-CA": "toggle English",
          "fr-CA": "toggle French"
        }
      }
    },
    "rich-text-key-value": {
      "sys": {
        "id": "rich-text-key-value",
        "space": {},
        "contentType": {
          "id": "keyRichText"
        }
      },
      "fields": {
        "slug": {
          "en-CA": "rich-text-key-slug"
        },
        "richText": {
          "en-CA": {
            "nodeType": "document",
            "data": {},
            "content": [
              {
                "nodeType": "paragraph",
                "data": {},
                "content": [
                  { "nodeType": "text", "data": {}, "value": "some text value" }
                ]
              }
            ]
          }
        }
      }
    },
    "some-complex-block-1": {
      "sys": {
        "id": "some-complex-block-1",
        "space": {},
        "contentType": {
          "id": "complexBlock"
        }
      },
      "fields": {
        "title": {
          "en-CA": "complex block title",
          "fr-CA": "complex block title - fr"
        },
        "body": {
          "en-CA": {
            "nodeType": "document",
            "data": {},
            "content": [
              {
                "nodeType": "paragraph",
                "data": {},
                "content": [
                  {
                    "nodeType": "text",
                    "data": {},
                    "value": "some paragraph in the complex block"
                  }
                ]
              }
            ]
          }
        }
      }
    },
    "some-complex-block-2": {
      "sys": {
        "id": "some-complex-block-2",
        "space": {},
        "contentType": {
          "id": "complexBlock"
        }
      },
      "fields": {
        "title": {
          "en-CA": "complex block 2 title",
          "fr-CA": "complex block 2 title - fr"
        },
        "body": {
          "en-CA": {
            "nodeType": "document",
            "data": {},
            "content": [
              {
                "nodeType": "paragraph",
                "data": {},
                "content": [
                  {
                    "nodeType": "text",
                    "data": {},
                    "value": "some paragraph in the complex block 2"
                  }
                ]
              }
            ]
          }
        }
      }
    },
    "some-block-group": {
      "sys": {
        "id": "some-block-group",
        "space": {},
        "contentType": {
          "id": "modularLayout"
        }
      },
      "fields": {
        "blocks": {
          "en-CA": [
            {
              "sys": {
                "type": "Link",
                "linkType": "Entry",
                "id": "some-complex-block-1"
              }
            },
            {
              "sys": {
                "type": "Link",
                "linkType": "Entry",
                "id": "some-complex-block-2"
              }
            }
          ]
        }
      }
    }
  },
  "linkedAssets": {
    "test-img.png": {
      "sys": {
        "id": "test-img.png"
      },
      "fields": {
        "title": {
          "en-CA": "some title",
          "fr-CA": "some title fr"
        },
        "description": {
          "en-CA": "some description",
          "fr-CA": "some description fr"
        },
        "file": {
          "en-CA": {
            "url": "/test-img.png",
            "fileName": "filename.png",
            "contentType": "image/png",
            "details": {
              "image": {
                "height": 20,
                "width": 20
              }
            }
          }
        }
      }
    },
    "test-img2.png": {
      "sys": {
        "id": "test-img2.png"
      },
      "fields": {
        "file": {
          "en-CA": {
            "url": "/test-img2.png",
            "fileName": "filename2.png",
            "contentType": "image/png",
            "details": {
              "image": {
                "height": 20,
                "width": 20
              }
            }
          }
        }
      }
    }
  }
}
