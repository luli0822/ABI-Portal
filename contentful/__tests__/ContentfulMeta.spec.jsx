import React from 'react';
import { shallow } from 'enzyme';
import ContentfulMeta from '../ContentfulMeta';
import {
	mockLang,
	mockPathMap,
	mockLinkedEntries,
	mockLinkedAssets,
	mockSiteEntry,
	mockFieldOrFallback,
} from '../__mocks__/SiteDataContext';
import { mockPageMeta, mockSiteMeta, mockNoFavicon } from '../__mocks__/ContentfulMeta';

const mockSiteEntryFn = jest.fn();
mockSiteEntryFn.mockReturnValue(mockSiteEntry);

jest.mock('react', () => {
	const ActualReact = require.requireActual('react');
	return {
		...ActualReact,
		useContext: () => ({
			pathMap: mockPathMap,
			fieldOrFallback: mockFieldOrFallback,
			linkedEntries: mockLinkedEntries,
			linkedAssets: mockLinkedAssets,
			siteEntry: mockSiteEntryFn(),
			lang: mockLang,
		}),
	};
});

describe('<ContentfulMeta />', () => {
	it('renders page title correctly', () => {
		// eslint-disable-next-line react/jsx-props-no-spreading
		expect(shallow(<ContentfulMeta {...mockPageMeta} />).debug()).toMatchSnapshot();
	});
	it('renders metaTitle prop correctly', () => {
		expect(
			// eslint-disable-next-line react/jsx-props-no-spreading
			shallow(<ContentfulMeta {...mockSiteMeta} metaTitle="meta title" />).debug()
		).toMatchSnapshot();
	});
	it('renders correctly with icons', () => {
		// eslint-disable-next-line react/jsx-props-no-spreading
		expect(shallow(<ContentfulMeta {...mockSiteMeta} />).debug()).toMatchSnapshot();
	});
	it('renders correctly with no icons', () => {
		mockSiteEntryFn.mockReturnValue(mockNoFavicon);
		// eslint-disable-next-line react/jsx-props-no-spreading
		expect(shallow(<ContentfulMeta {...mockSiteMeta} />).debug()).toMatchSnapshot();
	});
});
