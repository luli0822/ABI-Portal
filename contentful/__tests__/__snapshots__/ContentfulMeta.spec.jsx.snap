// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ContentfulMeta /> renders correctly with icons 1`] = `
"<Head>
  <title>
    en title
  </title>
  <meta name=\\"description\\" content=\\"description en\\" />
  <meta property=\\"og:url\\" content=\\"https://www.test-fr.ca///\\" />
  <link rel=\\"canonical\\" href=\\"https://www.test-fr.ca///\\" />
  <meta property=\\"og:description\\" content=\\"openGraphDescription en\\" />
  <meta property=\\"og:title\\" content=\\"openGraphTitle en\\" />
  <link rel=\\"icon\\" type=\\"image/png\\" href=\\"/favicon.png?fm=png\\" />
  <link rel=\\"icon\\" type=\\"image/png\\" sizes=\\"192x192\\" href=\\"/favicon.png?w=192&fm=png\\" />
  <link rel=\\"icon\\" type=\\"image/png\\" sizes=\\"96x96\\" href=\\"/favicon.png?w=96&fm=png\\" />
  <link rel=\\"icon\\" type=\\"image/png\\" sizes=\\"32x32\\" href=\\"/favicon.png?w=32&fm=png\\" />
  <link rel=\\"icon\\" type=\\"image/png\\" sizes=\\"16x16\\" href=\\"/favicon.png?w=16&fm=png\\" />
  <link rel=\\"apple-touch-icon\\" sizes=\\"180x180\\" href=\\"/favicon.png?w=180&fm=png\\" />
  <link rel=\\"apple-touch-icon\\" sizes=\\"152x152\\" href=\\"/favicon.png?w=152&fm=png\\" />
  <link rel=\\"apple-touch-icon\\" sizes=\\"144x144\\" href=\\"/favicon.png?w=144&fm=png\\" />
  <link rel=\\"apple-touch-icon\\" sizes=\\"114x114\\" href=\\"/favicon.png?w=114&fm=png\\" />
  <link rel=\\"apple-touch-icon\\" sizes=\\"120x120\\" href=\\"/favicon.png?w=120&fm=png\\" />
  <link rel=\\"apple-touch-icon\\" sizes=\\"76x76\\" href=\\"/favicon.png?w=76&fm=png\\" />
  <link rel=\\"apple-touch-icon\\" sizes=\\"60x60\\" href=\\"/favicon.png?w=60&fm=png\\" />
  <link rel=\\"apple-touch-icon\\" sizes=\\"57x57\\" href=\\"/favicon.png?w=57&fm=png\\" />
  <meta name=\\"msapplication-TileImage\\" content=\\"/favicon.png?fm=png\\" />
  <meta name=\\"apple-mobile-web-app-title\\" content=\\"en title\\" />
</Head>"
`;

exports[`<ContentfulMeta /> renders correctly with no icons 1`] = `
"<Head>
  <title>
    en title
  </title>
  <meta name=\\"description\\" content=\\"description en\\" />
  <meta property=\\"og:url\\" content=\\"https://www.test-fr.ca///\\" />
  <link rel=\\"canonical\\" href=\\"https://www.test-fr.ca///\\" />
  <meta property=\\"og:description\\" content=\\"openGraphDescription en\\" />
  <meta property=\\"og:title\\" content=\\"openGraphTitle en\\" />
  <meta name=\\"apple-mobile-web-app-title\\" content=\\"en title\\" />
</Head>"
`;

exports[`<ContentfulMeta /> renders metaTitle prop correctly 1`] = `
"<Head>
  <title>
    meta title
  </title>
  <meta name=\\"description\\" content=\\"description en\\" />
  <meta property=\\"og:url\\" content=\\"https://www.test-fr.ca///\\" />
  <link rel=\\"canonical\\" href=\\"https://www.test-fr.ca///\\" />
  <meta property=\\"og:description\\" content=\\"openGraphDescription en\\" />
  <meta property=\\"og:title\\" content=\\"openGraphTitle en\\" />
  <link rel=\\"icon\\" type=\\"image/png\\" href=\\"/favicon.png?fm=png\\" />
  <link rel=\\"icon\\" type=\\"image/png\\" sizes=\\"192x192\\" href=\\"/favicon.png?w=192&fm=png\\" />
  <link rel=\\"icon\\" type=\\"image/png\\" sizes=\\"96x96\\" href=\\"/favicon.png?w=96&fm=png\\" />
  <link rel=\\"icon\\" type=\\"image/png\\" sizes=\\"32x32\\" href=\\"/favicon.png?w=32&fm=png\\" />
  <link rel=\\"icon\\" type=\\"image/png\\" sizes=\\"16x16\\" href=\\"/favicon.png?w=16&fm=png\\" />
  <link rel=\\"apple-touch-icon\\" sizes=\\"180x180\\" href=\\"/favicon.png?w=180&fm=png\\" />
  <link rel=\\"apple-touch-icon\\" sizes=\\"152x152\\" href=\\"/favicon.png?w=152&fm=png\\" />
  <link rel=\\"apple-touch-icon\\" sizes=\\"144x144\\" href=\\"/favicon.png?w=144&fm=png\\" />
  <link rel=\\"apple-touch-icon\\" sizes=\\"114x114\\" href=\\"/favicon.png?w=114&fm=png\\" />
  <link rel=\\"apple-touch-icon\\" sizes=\\"120x120\\" href=\\"/favicon.png?w=120&fm=png\\" />
  <link rel=\\"apple-touch-icon\\" sizes=\\"76x76\\" href=\\"/favicon.png?w=76&fm=png\\" />
  <link rel=\\"apple-touch-icon\\" sizes=\\"60x60\\" href=\\"/favicon.png?w=60&fm=png\\" />
  <link rel=\\"apple-touch-icon\\" sizes=\\"57x57\\" href=\\"/favicon.png?w=57&fm=png\\" />
  <meta name=\\"msapplication-TileImage\\" content=\\"/favicon.png?fm=png\\" />
  <meta name=\\"apple-mobile-web-app-title\\" content=\\"en title\\" />
</Head>"
`;

exports[`<ContentfulMeta /> renders page title correctly 1`] = `
"<Head>
  <title>
    title en | en title
  </title>
  <meta name=\\"description\\" content=\\"description en\\" />
  <meta property=\\"og:url\\" content=\\"https://www.test.ca/about-pg-en/\\" />
  <link rel=\\"canonical\\" href=\\"https://www.test.ca/about-pg-en/\\" />
  <meta property=\\"og:description\\" content=\\"openGraphDescription en\\" />
  <meta property=\\"og:title\\" content=\\"openGraphTitle en\\" />
  <link rel=\\"icon\\" type=\\"image/png\\" href=\\"/favicon.png?fm=png\\" />
  <link rel=\\"icon\\" type=\\"image/png\\" sizes=\\"192x192\\" href=\\"/favicon.png?w=192&fm=png\\" />
  <link rel=\\"icon\\" type=\\"image/png\\" sizes=\\"96x96\\" href=\\"/favicon.png?w=96&fm=png\\" />
  <link rel=\\"icon\\" type=\\"image/png\\" sizes=\\"32x32\\" href=\\"/favicon.png?w=32&fm=png\\" />
  <link rel=\\"icon\\" type=\\"image/png\\" sizes=\\"16x16\\" href=\\"/favicon.png?w=16&fm=png\\" />
  <link rel=\\"apple-touch-icon\\" sizes=\\"180x180\\" href=\\"/favicon.png?w=180&fm=png\\" />
  <link rel=\\"apple-touch-icon\\" sizes=\\"152x152\\" href=\\"/favicon.png?w=152&fm=png\\" />
  <link rel=\\"apple-touch-icon\\" sizes=\\"144x144\\" href=\\"/favicon.png?w=144&fm=png\\" />
  <link rel=\\"apple-touch-icon\\" sizes=\\"114x114\\" href=\\"/favicon.png?w=114&fm=png\\" />
  <link rel=\\"apple-touch-icon\\" sizes=\\"120x120\\" href=\\"/favicon.png?w=120&fm=png\\" />
  <link rel=\\"apple-touch-icon\\" sizes=\\"76x76\\" href=\\"/favicon.png?w=76&fm=png\\" />
  <link rel=\\"apple-touch-icon\\" sizes=\\"60x60\\" href=\\"/favicon.png?w=60&fm=png\\" />
  <link rel=\\"apple-touch-icon\\" sizes=\\"57x57\\" href=\\"/favicon.png?w=57&fm=png\\" />
  <meta name=\\"msapplication-TileImage\\" content=\\"/favicon.png?fm=png\\" />
  <meta name=\\"apple-mobile-web-app-title\\" content=\\"en title\\" />
</Head>"
`;
